### PR TITLE
[MERGE FIRST] Cancel pods 

### LINKED ISSUE #53: Support Cancelling Jobs
Although cancelling will be done by timeout, since the domain containers the job-id, there is no reason why we cannot allow users to cancel a job by simply pasting the URL. We can then extract the correct part and cancel the job.

### PR DESCRIPTION
**Pull request by Sharpz7**
Closes #53 

Currently need advice on testing `test_cancel()` within `test_typer.py`

### CODE REVIEW

File: armada_jupyter/__main__.py [hunk 1]
```diff
@@ -30,6 +30,16 @@ def submit(file: str):
     submit_worker(file, armada_client)
 
 
+@app.command()
+def cancel(url: str):
+    """
+    Accepts a URL and cancels the corresponding job
```
Comment (kannon92) [inline armada_jupyter/__main__.py:36]: The docs don't make it really clear what URL we expect?  I guess it must be the lookout url plus armada?

File: armada_jupyter/armada_utils.py [hunk 2]
```diff
@@ -131,3 +132,21 @@ def check_job_status(client: ArmadaClient, submission: Submission, job_id: str)
...
+
+    # regex to find anything between armada- and -0
+    result = re.search(r"armada-(.+?)-0", url)
+
+    if result is None:
+        raise Exception("Could not find job_id in URL")
+
+    job_id = result.group(1)
+
+    client.cancel_jobs(job_id=job_id)
```
Comment (kannon92) [inline armada_jupyter/armada_utils.py:150]: What if cancel fails?

File: docs/quickstart.md [hunk 3]
```diff
@@ -50,10 +50,10 @@ You can access the JupyterLab instance by using the URL's that are printed out b
...
 
-## Cancelling all Jobs
+## Cancelling a Job
 
-**NOTE: This will kill all jobs running on queue `default` and job-set-id `testing`**
+To cancel a job you can use the following command:
 
 ```
-python3 ./docs/dev/cancel.py
+python3 -m armada_jupyter cancel {URL}
```
Comment (kannon92) [inline docs/quickstart.md:58]: For docs and readme, I would like to see what this URL should be?

**Review by kannon92 — COMMENTED** [review]
It is looking good.  Minor fixes around doc fixes around URL and handling cases if cancel errors.

File: docs/quickstart.md (hunk 3 above)
Comment (Sharpz7) [reply docs/quickstart.md:58]: Yeah. I definitely need to document this better. Thanks

**Review by kannon92 — APPROVED** [review]

**Pull request merged**